src/D2L.CodeStyle.Analyzers/Immutability/ImmutableAttributeConsistencyChecker.cs

```diff
@@ -45,7 +46,55 @@ INamedTypeSymbol typeSymbol
 			}
 		}
 
-		public void CompareConsistencyToBaseType(
+		public void CheckMethodDeclaration(
+			IMethodSymbol methodSymbol
+		) {
+			if( methodSymbol.TypeParameters.Length == 0 ) {
+				return;
+			}
+
+			MethodDeclarationSyntax methodSyntax = null;
+			Location[] parameterLocations = null;
+			Location GetLocationOfNthParameter( int N ) {

```

Comment:
Can you just make this a private method?

These methods generate extra code behind the scene and do heap allocations to deal with the captures.

It's simpler, faster and more idiomatic to use a private helper method.

Reply:
The local function is nice if you need to return a `Func<...>` or something.

Reply:
Had it there for caching, but I'll just remove it since perf doesn't really matter in error cases.